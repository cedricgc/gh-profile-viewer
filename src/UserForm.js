import React, { Component } from 'react';

class UserForm extends Component {
  constructor(props) {
    super(props);
    this.state = { username: '' };

    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  handleChange(event) {
    this.setState({ username: event.target.value });
  }

  handleSubmit(event) {
    event.preventDefault();
  }

  render() {
    return (
      <div>
        <form onSubmit={this.handleSubmit}>
          <h2>Input the Github user to display</h2>
          <section>
            <div>
              <label>
                <span>Username: </span>
                <input
                  type="text"
                  id="username"
                  name="username"
                  data-test-id="username"
                  onChange={this.handleChange}
                />
              </label>
              <input type="submit" value="submit" />
            </div>
          </section>
        </form>
      </div>
    );
  }
}

export default UserForm;
