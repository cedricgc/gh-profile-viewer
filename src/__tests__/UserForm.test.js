import React from 'react';
import ReactTestRenderer from 'react-test-renderer';

import UserForm from '../UserForm';

test('component matches snapshot', () => {
  const render = ReactTestRenderer.create(
    <UserForm onUsernameSubmit={(user) => {}} />
  );

  const tree = render.toJSON();
  expect(tree).toMatchSnapshot();
});
