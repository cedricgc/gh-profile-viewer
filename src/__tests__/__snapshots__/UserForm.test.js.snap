// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component matches snapshot 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <h2>
      Input the Github user to display
    </h2>
    <section>
      <div>
        <label>
          <span>
            Username: 
          </span>
          <input
            data-test-id="username"
            id="username"
            name="username"
            onChange={[Function]}
            type="text"
          />
        </label>
        <input
          type="submit"
          value="submit"
        />
      </div>
    </section>
  </form>
</div>
`;
